src/palette/internal/palettelayout.cpp

```diff
@@ -129,3 +132,53 @@ void PaletteLayout::layout(Clef* item, const Context& ctx)
     RectF bbox = item->symBbox(item->symId());
     item->setbbox(bbox);
 }
+
+void PaletteLayout::layout(KeySig* item, const Context& ctx)
+{
+    double spatium = item->spatium();
+    double step = spatium * 0.5;
+
+    item->setbbox(RectF());
+    item->keySymbols().clear();
+
+    Key key = item->key();
+    const signed char* lines = ClefInfo::lines(ClefType::G);
+
+    if (std::abs(int(key)) <= 7) {
+        SymId sym = key > 0 ? SymId::accidentalSharp : SymId::accidentalFlat;
+        double accidentalGap = ctx.style().styleS(Sid::keysigAccidentalDistance).val();
+        double previousWidth = item->symWidth(sym) / spatium;
+        int lineIndexOffset = int(key) > 0 ? 0 : 7;
+        for (int i = 0; i < std::abs(int(key)); ++i) {
+            int line = lines[lineIndexOffset + i];
+            KeySym ks;
+            ks.sym = sym;
+            double x = 0.0;
+            if (item->keySymbols().size() > 0) {
+                const KeySym& previous = item->keySymbols().back();
+                x = previous.xPos + previousWidth + accidentalGap;
+                bool isAscending = line < previous.line;
+                SmuflAnchorId currentCutout = isAscending ? SmuflAnchorId::cutOutSW : SmuflAnchorId::cutOutNW;
+                SmuflAnchorId previousCutout = isAscending ? SmuflAnchorId::cutOutNE : SmuflAnchorId::cutOutSE;
+                PointF cutout = item->symSmuflAnchor(sym, currentCutout);
+                double currentCutoutY = line * step + cutout.y();
+                double previousCutoutY = previous.line * step + item->symSmuflAnchor(previous.sym, previousCutout).y();
+                if ((isAscending && currentCutoutY < previousCutoutY) || (!isAscending && currentCutoutY > previousCutoutY)) {
+                    x -= cutout.x() / spatium;
+                }
+            }
+            ks.xPos = x;
+            ks.line = line;
+            item->keySymbols().push_back(ks);
+        }
+    } else {
+        LOGD() << "illegal key:" << int(key);
```

Comment:
Does this mean that it isn't possible to layout custom key signatures in the palette? It must be possible, because custom key signatures can be added to the palette using the "Create key signature" button from the "More" popup in the key sig palette.

Reply:
Thank you, I will fix it

Reply:
Added layout for custom keysig